lib/livebook/zta/livebook_teams.ex

```diff
@@ -103,7 +103,12 @@ defmodule Livebook.ZTA.LivebookTeams do
   defp request_user_authentication(conn, team) do
     case Teams.Requests.create_auth_request(team) do
       {:ok, %{"authorize_uri" => authorize_uri}} ->
-        current_url = LivebookWeb.Endpoint.url() <> conn.request_path <> "?teams_identity"
+        current_url =
+          conn
+          |> Plug.Conn.request_url()
+          |> URI.parse()
+          |> URI.append_query("teams_identity")
+          |> URI.to_string()
```

Comment:
I am worried this may not be enough when deployed behind a proxy, such as on Fly. This is because when talking to Livebook, the host will be some internal IP used by Fly. We would need to tell people to set `LIVEBOOK_PROXY_HEADERS="x-forwarded-for"`. We could also automatically look for this header and build the URL by hand using the proxy headers or the connection information.

Reply:
I checked and the Host will be correct, but not the port and the schema. I think it will just work on Fly by luck but we may need to document this clearly.

Reply:
@josevalim maybe I should add the `proxy_set_header X-Forwarded-Port` config [here](https://github.com/livebook-dev/livebook/blob/main/docs/deployment/nginx_https.md?plain=1#L33) as well?